File: ETC/PLATFORM/프로그래머스/2/42584. 주식가격/주식가격.py
#초 단위로 주식가격이 기록 됨. prices
#가격이 떨어지지 않은 기간은 몇초
#prices=[1,2,3,2,3]
def solution(prices):
    answer = []
    for i in range(len(prices)):
        cnt=0
        for j in range(i+1,len(prices)):
            cnt+=1
            if(prices[i] > prices[j]):
                break
        answer.append(cnt)

    return answer

print(solution([1,2,3,2,3]))

File: ETC/PLATFORM/프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public Queue<Integer> solution(int[] prices) {
        Queue<Integer> que = new LinkedList<>();
        Queue<Integer> answer = new LinkedList<>();

        for (int i = 0; i < prices.length; i++) {
            que.add(prices[i]);
        }

        while (que.size() > 1) {  // 마지막 원소 전까지 처리
            int v = que.poll();
            int undrop = 0;

            for (int a : que) {
                undrop++;
                if (v > a) break;  // 떨어지면 멈춤
            }
            answer.add(undrop);
        }

        answer.add(0);  // 마지막 가격은 비교할 대상이 없으니 0초 유지

        return answer;
    }
}
